.github/workflows/helm-ci.yml

```diff
@@ -18,10 +18,12 @@ jobs:
   conftest:
     runs-on: ubuntu-latest
     container:
-      image: grafana/mimir-build-image:pr3976-e7cae18e3
+      image: grafana/mimir-build-image:goupdate-72d66708c
     steps:
       - name: Checkout repository
         uses: actions/checkout@v2
+        run: git config --global --add safe.directory /__w/mimir/mimir
```

Comment:
Why are we doing these changes? Seems unrelated to Go update.

Reply:
The pipeline failed with dubious ownership in repository, i have seen this [Issue](https://github.com/actions/runner-images/issues/6775), that says the latest ubuntu image activate the security patch which cause that issue. Most of people solved issue by adding the git config command. But I wonder why other PR are not impacted

Reply:
Is that related to using `actions/checkout@v3`? Can you please update this PR description why the additional changes are needed?

Reply:
I think the error is triggered by the Go compiler `-buildvcs` flag, but I'm not sure if it changed in 1.20 or what.

Reply:
I believe [this](https://github.com/actions/checkout/issues/1169) is also the root issue.

Reply:
@pstibrany Sorry I move the PR to draft, the pipeline is still not passing, I am trying different things. Once it is passed I would summarise the findings and separate what is needed to change in github action and the go version update into 2 PRs with a better description.